hw/ip/otbn/dv/smoke/smoke_test.s

```diff
@@ -5,7 +5,7 @@
 # OTBN Smoke test, runs various instructions which are expected to produce the
 # final register state see in smoke_expected.txt
 
-.section .text
+.section .text.start
```

Comment:
I don't think I quite understand this change. The existing `.section` lines plonked everything into `.text`. Since it's a single compilation unit, this first instruction ended up at the location that you're getting from `.text.start`.

Why does this need to change? And if it does need to change, is there any reason to add the `.text` bit lower down? I'd have thought that everything could go into `.text.start`...

Reply:
The `.text.start` is needed because `.text` is not a GC root when using `hw/ip/otbn/data/otbn.ld.tpl`. This wasn't needed before because the linking in the DV test doesn't make use of garbage collection.

Yes the second is `.text` is a little superfluous; I'll remove it.

Reply:
Thanks for explaining. This looks sensible to me.